### PR TITLE
DB 5.1__0

### PR DESCRIPTION
**Pull request by joshmoore**
This is primarily a renaming of the DEV20 DB to the final release name but does include:

 * a minor fix to the pressure unit, renaming `kBar` to `kbar`
 * a widening of the map annotation type
 * an addition of a `lookupTable` column to channel and channelbinding
 * a restriction to PG >= 9.2
 * a bump to the PG JDBC driver

cc: @mtbc @qidane

### CODE REVIEW

**Comment by mtbc** [comment]
We probably also want to get the PostgreSQL version number check in given the new prerequisite.

**Comment by joshmoore** [comment]
Pushed the 9.2 restriction. Am I right that `840000` was a typo? I've set it to `902000` based on my local results:
```
ome6=#  SELECT CAST(setting AS INTEGER) version_num FROM pg_settings WHERE name = 'server_version_num';
 version_num
-------------
       90305
(1 row)

Time: 6.677 ms
```

**Comment by mtbc** [comment]
Argh, yes, a typo, well-spotted.

**Comment by mtbc** [comment]
Also need version check for script from patch 20?

**Comment by joshmoore** [comment]
> Also need version check for script from patch 20?

* https://github.com/openmicroscopy/openmicroscopy/pull/3519/files#diff-346c7b614f0229c3ad2d72c32712f679R42
 * https://github.com/openmicroscopy/openmicroscopy/pull/3519/files#diff-3a3d57c3b7a70f5912ef97a264b96336R42

or am I missing something?

**Comment by mtbc** [comment]
I was thinking that `sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql` needs the,
```sql
SELECT assert_db_server_prerequisites(90200);
```
and associated stuff.

**Comment by joshmoore** [comment]
> needs the SELECT assert_db_server_prerequisites(90200);

Ah, I see what you mean. Pushed.

**Comment by mtbc** [comment]
Thank you, looks good. (I was a bit vague!)

**Comment by mtbc** [comment]
Latest commit looks fine.

**Comment by joshmoore** [comment]
Still waiting on the new VM image to get the queue green. I'll probably add one other change, too.

**Comment by mtbc** [comment]
Looks plausible, we can rethink the repesentation for 5.2; I'm guessing there's a card.

(I don't know if it's worth building something deeply into server code to make sure that only valid four-int text is used in this property but, especially with the code generation, maybe that's just too tricky to be worth it.)

**Comment by sbesson** [comment]
From Blitz.log of https://ci.openmicroscopy.org/view/Failing/job/OMERO-5.1-breaking-deploy/249/console

```
DB version (OMERO5.1DEV__20) does not match omero.properties (OMERO5.1__0). Please apply a db upgrade.
See http://www.openmicroscopy.org/site/support/omero5/sysadmins/server-upgrade.html
```

**Comment by joshmoore** [comment]
Thanks, @sbesson. Pushed.

File: sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+--
+-- Actual upgrade
+--
+
+UPDATE pg_enum
+    SET enumlabel = 'kbar'
+    WHERE enumlabel = 'kBar' AND enumtypid = (
+        SELECT oid FROM pg_type WHERE typname = 'unitspressure'
+    );
```
Comment (sbesson) [inline sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql]: From https://ci.openmicroscopy.org/job/OMERO-5.1-breaking-upgrade/VERSION=LAST_DEV,label=trout/259/console


```
2015-02-27 23:14:58,578 [    omego.db] INFO  Upgrading database using ./OMERO.server-5.1.0-m4-211-6096272-ice34-b305/sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql
2015-02-27 23:14:58,578 [omego.extern] INFO  Executing [custom environment]: psql -d OMERO-5.1-breaking-upgrade -h localhost -U omero -w -A -t -f ./OMERO.server-5.1.0-m4-211-6096272-ice34-b305/sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql
2015-02-27 23:14:58,595 [    omego.db] WARNI stderr: psql:./OMERO.server-5.1.0-m4-211-6096272-ice34-b305/sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql:56: ERROR:  permission denied for relation pg_enum
psql:./OMERO.server-5.1.0-m4-211-6096272-ice34-b305/sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql:80: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:./OMERO.server-5.1.0-m4-211-6096272-ice34-b305/sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql:82: ERROR:  current transaction is aborted, commands ignored until end of transaction block
```

Comment (joshmoore) [inline sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql]: Hmmm..... don't know if I'll be able to find a solution that will work as a non-`superuser`. Will take some looking into. As an alternative, I could just add the extra `kbar` and we live with the duplication (under the assumption that no one is using it anyway...)

Comment (sbesson) [inline sql/psql/OMERO5.1__0/OMERO5.1DEV__20.sql]: Fair enough as this corresponds to the development patches which are not meant to be used in production ugprade (and will likely be deleted anyways?).
Let's make sure we throughly check the production upgrade script works as expected wrt this `enumlabel`

**Comment by joshmoore** [comment]
Pushed a commit which simply leaves kbar and kBar in the dev line.

**Comment by sbesson** [comment]
https://ci.openmicroscopy.org/job/OMERO-5.1-breaking-upgrade/262/ looks good. Removing breaking label before this morning rebuild.

**Comment by sbesson** [comment]
Merging for 5.1.0-m5

**Pull request merged**